### PR TITLE
Fixes #877: Apps in discovery view should link to new details page

### LINKED ISSUE #877: Apps in discovery view should link to new details page
Use the new details page link instead of the `app.url` link. Depends on #868

Code here: https://github.com/mozilla/webmaker-app/blob/master/views/discover/index.html

### PR DESCRIPTION
**Pull request by gesa**

### CODE REVIEW

**Comment by k88hudson** [comment]
@gesa arg sorry, should have been more clear -- this should be applied to the `/discovery` view, not the profile view

**Comment by gesa** [comment]
oh @k88hudson i'm sorry, it was clear. I made assumptions.

So, there are two designs here:

1. https://redpen.io/vd06771345f2712a65 (which wasn't linked in #868) is clearly meant for being linked to in the discovery view.
2. https://redpen.io/ck555642cf38b02adb (linked in #868) is _presumably_ meant for being linked to in the profile view?

I'm unclear on which to implement. If the answer is both, I'd ❤️ some pairing time to help figure out how to make that work.

**Comment by k88hudson** [comment]
You know what, I think you're right on both points. Happy to discuss how to do this next week.

**Comment by xmatthewx** [comment]
:+1:  landing on a details view for all apps all the time. 

 - From a user's perspective, 1 & 2 from @gesa's list above are the same view, with extra stuff added if you're the app's owner. 
 - From our perspective, these could be 2 views, or 1 view with extra stuff for app owners -- whichever is easier to build and maintain. 

(Forgive me if I'm misunderstanding this discussion.)

**Comment by gesa** [comment]
No you totally hit it on the head, @xmatthewx:exclamation: Thanks for the insight. I'm not :100:% certain how that will be implemented, but i will plan to pair with @k88hudson (ooooor, maybe even @gvn?) next week to figure out the best way to pull it off.

(sorry for the emoji overload)

(wait no i'm totally not.) :smiley_cat:

File: views/discover/index.html [hunk 1]
```diff
@@ -5,11 +5,11 @@
 </div>
 
 <ul class="list-cell" v-show="mode === 'Featured'">
-    <li v-component="appCell" v-repeat="app: apps.featured" app-id="{{ app.url }}"></li>
+    <li v-component="appCell" v-repeat="app: apps.featured" app-id="{{ app.url }}" suffix="detail/"></li>
```
Comment (k88hudson) [inline views/discover/index.html]: you need to remove the app-id property now since we need relative links

File: components/appCell/index.js [hunk 2]
```diff
@@ -13,7 +13,7 @@ module.exports = {
     data: {
```
Comment (k88hudson) [inline components/appCell/index.js]: you need to add the suffix in the constructedUrl construction

File: components/appCell/index.js [hunk 3]
```diff
@@ -13,7 +13,7 @@ module.exports = {
...
         constructedURL: ''
     },
-    paramAttributes: ['prefix', 'app-id'],
+    paramAttributes: ['prefix', 'app-id', 'suffix'],
```
Comment (k88hudson) [inline components/appCell/index.js]: remove app-id

File: views/discover/index.html [hunk 4]
```diff
@@ -5,11 +5,11 @@
...
 </ul>
 
 <ul class="list-cell" v-show="mode === 'Newest'">
-    <li v-component="appCell" v-repeat="app: apps.newest" app-id="{{ app.url }}"></li>
+    <li v-component="appCell" v-repeat="app: apps.newest" app-id="{{ app.url }}" suffix="detail/"></li>
```
Comment (k88hudson) [inline views/discover/index.html]: /detail/

**Comment by k88hudson** [comment]
Just a few fixes :+1:

File: views/discover/index.html (hunk 4 above)
Comment (gesa) [inline views/discover/index.html]: The links I had in the discover section ended with a forward slash

**Comment by k88hudson** [comment]
@gesa i lied, i think you might need to use `$index` instead

**Comment by k88hudson** [comment]
Woooot :tada: :tada: :tada: :tada: :tada: :tada:

**Pull request merged**